push/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java

```diff
@@ -150,17 +150,18 @@ private void cleanupInvalidRegistrationIDsForVariant(String variantID, Multicast
         // read the results:
         for (int i = 0; i < results.size(); i++) {
             // use the current index to access the individual results
-            Result result = results.get(i);
+            final Result result = results.get(i);
 
-            // is there an error code that indicates an invalid regID ?
-            if (Constants.ERROR_INVALID_REGISTRATION.equals(result.getErrorCodeName())) {
+            final String errorCodeName = result.getErrorCodeName();
+            // is there an error code that indicates an invalid or not registered registration ID ?
+            if (Constants.ERROR_INVALID_REGISTRATION.equals(errorCodeName) || Constants.ERROR_NOT_REGISTERED.equals(errorCodeName)) {
```

Comment:
@secondsun good idea - makes it also easier to update, based on future error codes